internal/app/api/handler/friend/friend.go

```diff
@@ -40,3 +47,75 @@ func (h *Handler) Get(w http.ResponseWriter, r *http.Request) {
 	}
 	respond.JSON(w, http.StatusOK, friend)
 }
+
+// GetAll handle get gifts HTTP Request
+func (h *Handler) GetAll(w http.ResponseWriter, r *http.Request) {
+	friends, err := h.srv.GetAll(r.Context())
+	if err != nil {
+		respond.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+	
+	respond.JSON(w, http.StatusOK, friends)
+
+}
+
+// Create handle insert gift HTTP Request
+func (h *Handler) Create(w http.ResponseWriter, r *http.Request) {
+	var friend types.Friend
+
+	if (r.Body == nil) {
```

Comment:
delete this block of code. Move the line `defer r.Body.Close() ` to line #81

Reply:
Fixed